freqtrade/tests/edge/test_edge.py

```diff
@@ -0,0 +1,275 @@
+
+    assert edge.calculate()
+    assert len(edge._cached_pairs) == 2
+    assert edge._last_updated <= arrow.utcnow().timestamp + 2
+
+
+def test_process_expectancy(mocker, default_conf):
+    default_conf['edge']['min_trade_number'] = 2
+    exchange = get_patched_exchange(mocker, default_conf)
+
+    def get_fee():
+        return 0.001
+
+    exchange.get_fee = get_fee
+    edge = Edge(default_conf, exchange)
+
+    trades = [
+        {'pair': 'TEST/BTC',
+         'stoploss': -0.9,
+         'profit_percent': '',
+         'profit_abs': '',
+         'open_time': np.datetime64('2018-10-03T00:05:00.000000000'),
+         'close_time': np.datetime64('2018-10-03T00:10:00.000000000'),
+         'open_index': 1,
+         'close_index': 1,
+         'trade_duration': '',
+         'open_rate': 17,
+         'close_rate': 17,
+         'exit_type': 'sell_signal'},  # sdfsdf
+
+        {'pair': 'TEST/BTC',
+         'stoploss': -0.9,
+         'profit_percent': '',
+         'profit_abs': '',
+         'open_time': np.datetime64('2018-10-03T00:20:00.000000000'),
+         'close_time': np.datetime64('2018-10-03T00:25:00.000000000'),
+         'open_index': 4,
+         'close_index': 4,
+         'trade_duration': '',
+         'open_rate': 20,
+         'close_rate': 20,
+         'exit_type': 'sell_signal'},
+
+        {'pair': 'TEST/BTC',
+         'stoploss': -0.9,
+         'profit_percent': '',
+         'profit_abs': '',
+         'open_time': np.datetime64('2018-10-03T00:30:00.000000000'),
+         'close_time': np.datetime64('2018-10-03T00:40:00.000000000'),
+         'open_index': 6,
+         'close_index': 7,
+         'trade_duration': '',
+         'open_rate': 26,
+         'close_rate': 34,
+         'exit_type': 'sell_signal'}
+    ]
+
+    trades_df = DataFrame(trades)
+    trades_df = edge._fill_calculable_fields(trades_df)
+    final = edge._process_expectancy(trades_df)
+    assert len(final) == 1
+
+    # TODO: check expectancy + win rate etc
+
+
+def test_remove_open_trade_at_the_end(mocker, default_conf):
+    exchange = get_patched_exchange(mocker, default_conf)
+    edge = Edge(default_conf, exchange)
+
+    stoploss = -0.99  # we don't want stoploss to be hit in this test
+    ticker = [
+        # D=Date, B=Buy,  O=Open,  H=High,  L=Low,  C=Close, S=Sell
+        # D, B,  O,  H,  L,  C, S
+        [3, 1, 12, 25, 11, 20, 0],  # ->
+        [4, 0, 20, 30, 19, 25, 1],  # -> should enter the trade
+    ]
+
+    ticker_df = _build_dataframe(ticker)
+    trades = edge._find_trades_for_stoploss_range(ticker_df, 'TEST/BTC', [stoploss])
+
+    # No trade should be found
+    assert len(trades) == 0
+
+
+def test_two_complete_trades(mocker, default_conf):
+    exchange = get_patched_exchange(mocker, default_conf)
+    edge = Edge(default_conf, exchange)
+
+    stoploss = -0.99  # we don't want stoploss to be hit in this test
+    ticker = [
+        # D=Date, B=Buy,  O=Open,  H=High,  L=Low,  C=Close, S=Sell
```

Comment:
i find the sequence of this data quite irritating - can't we have "Open, High, Low, Close, Volume" (a regular ticker dataset)- and then extend to the right with "buy" and "sell" ?

Reply:
totally agree, there is no need for a new uncommon template. going to correct it. thanks !

Reply:
working on aligning our 2 versions ... maybe have a look at the new temporary version here: 
https://github.com/xmatthias/freqtrade/blob/feat/functional_tests/freqtrade/tests/optimize/test_backtest_detail.py

Reply:
I am going to unify tests based on your version. either on this PR or once it is merged in another PR. I don't have any preference though ... tell me what you prefer !

Reply:
Now that I have thought about it more, I prefer to do another PR on that if you are ok. first going to complete other tasks then once your BT PR merged going to do a PR. is it ok ?

Reply:
lets make sure we don't forget ... ;)